sdk/core/azure-core/inc/azure/core/http/websockets/curl_websockets_transport.hpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief #Azure::Core::Http::HttpTransport implementation via CURL.
+ */
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/curl_transport.hpp"
+#include "azure/core/http/http.hpp"
+#include "azure/core/http/transport.hpp"
+#include "azure/core/http/websockets/websockets_transport.hpp"
+#include <memory>
+
+namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+
+  /**
+   * @brief Concrete implementation of a WebSocket Transport that uses libcurl.
+   */
+  class CurlWebSocketTransport : public CurlTransport, public WebSocketTransport {
+  public:
+    /**
+     * @brief Construct a new CurlWebSocketTransport object.
+     *
+     * @param options Optional parameter to override the default options.
+     */
+    CurlWebSocketTransport(CurlTransportOptions const& options = CurlTransportOptions())
```

Comment:
Is the expectation that WebSocket options would be in curltransportoptions?

Reply:
A `CurlWebSocketTransport` is a `CurlTransport`. And CurlTransport needs options.